pwem/protocols/protocol_set_editor.py

```diff
@@ -0,0 +1,282 @@
+                      choices=[self.LOCAL_SYM_NAME[SYM_I222] +
+                               " (" + SCIPION_SYM_NAME[SYM_I222] + ")",
+                               self.LOCAL_SYM_NAME[SYM_I222r] +
+                               " (" + SCIPION_SYM_NAME[SYM_I222r] + ")",
+                               self.LOCAL_SYM_NAME[SYM_In25] +
+                               " (" + SCIPION_SYM_NAME[SYM_In25] + ")",
+                               self.LOCAL_SYM_NAME[SYM_In25r] +
+                               " (" + SCIPION_SYM_NAME[SYM_In25r] + ")",
+                               self.LOCAL_SYM_NAME[SYM_I2n3] +
+                               " (" + SCIPION_SYM_NAME[SYM_I2n3] + ")",
+                               self.LOCAL_SYM_NAME[SYM_I2n3r] +
+                               " (" + SCIPION_SYM_NAME[SYM_I2n3r] + ")",
+                               self.LOCAL_SYM_NAME[SYM_I2n5] +
+                               " (" + SCIPION_SYM_NAME[SYM_I2n5] + ")",
+                               self.LOCAL_SYM_NAME[SYM_I2n5r] +
+                               " (" + SCIPION_SYM_NAME[SYM_I2n5r] + ")"
+                               ],
+                      default=SYM_I222r - SYM_I222,
+                      )
+
+    def _insertAllSteps(self):
+        operation =  self.operation.get()
+        if (operation == self.CHOICE_FORMULA):
+            self._insertFunctionStep('formulaStep')
+        elif (operation == self.CHOICE_ROTATE_VECTOR):
+            self._insertFunctionStep('rotateStep')
+        elif (operation == self.CHOICE_ROTATE_ICOSAHEDRAL):
+            self._insertFunctionStep('rotateIcoStep')
+
+
+    def rotateIcoStep(self):
+        """
+        Compute rotation matrix from one icosahedral
+        symmetry to another and apply it
+        to projection directions. Reconstructed volume
+        should change symmetry
+        """
+        from pwem.convert.symmetry import Icosahedron
+        origSym = self.originSymmetryGroup.get() + SYM_I222
+        targetSym = self.targetSymmetryGroup.get() + SYM_I222
+        ico = Icosahedron(orientation = SCIPION_SYM_NAME[origSym][1:])
+        matrix = ico.coordinateSystemTransform(SCIPION_SYM_NAME[origSym][1:],
+                                               SCIPION_SYM_NAME[targetSym][1:])
+        # convert to numpy array
+        # and add extra row
+        matrix =  np.array(matrix)
+        matrix = np.append(matrix, [[0, 0, 0, 1]], axis=0)
+
+        inputSet = self.inputSet.get()
+        modifiedSet = inputSet.createCopy(self._getExtraPath(), copyInfo=True)
+        for sourceItem in inputSet.iterItems():
+            item = sourceItem.clone()
+            transformation = item.getTransform()
+            transformation.composeTransform(matrix)
+            modifiedSet.append(item)
+        self.createOutput(self.inputSet, modifiedSet)
+
+    def createOutput(self, inputSet, modifiedSet):
+        # TODO: getExtendend is empty when protocol called
```

Comment:
There are 2 ways to define a pointer, a direct pointer, and an indirect one.

Through the gui is not possible to have a direct pointer (point directly to an id that is the output itself)
Only from tests we create direct pointers

One way is to create indrect pointers in the test like:

prot.inputSet.set(protWithOutput)
prot.inputSet.setExtended("outputSet")

Reply:
You want to keep tha same name as the input?